Updating RKE2 Restore Custom Cluster Steps (#1252)
rancher/rancher-docs: docs/how-to-guides/new-user-guides/backup-restore-and-disaster-recovery/restore-rancher-launched-kubernetes-clusters-from-backup.md

    1. In the **Clusters** page, go to the cluster where you want to remove nodes.
    1. In the **Machines** tab, click **⋮ > Delete** on each node you want to delete. Initially, you will see the nodes hang in a `deleting` state, but once all etcd nodes are deleting, they will be removed together. This is due to the fact that Rancher sees all etcd nodes deleting and proceeds to "short circuit" the etcd safe-removal logic.

> 1. After all etcd nodes are removed, add a new etcd node that you are planning to restore from. The new node must be assigned the role of `all` (etcd, controlplane, and worker).

Reviewer: ```suggestion
1. After all etcd nodes are removed, add the new etcd node that you are planning to restore from. Assign the new node to the role of `all` (etcd, controlplane, and worker).
```

Reviewer 2: `Assign the new node to the role` - the `to` needs to be dropped otherwise the meaning is changed (and incorrect).

Reviewer: Updated the edit suggestion

Reviewer: Updated the two others as well

Author: Updated with suggestion

>     - If the node has previously been used in a cluster, [clean the node](../manage-clusters/clean-cluster-nodes.md#cleaning-up-nodes) first.

Reviewer: ```suggestion
    - If the node was previously in a cluster, [clean the node](../manage-clusters/clean-cluster-nodes.md#cleaning-up-nodes) first.
```

    - For custom clusters, go to the **Registration** tab and check the box for `etcd, controlplane, and worker`. Then copy and run the registration command on your node.
    - For node driver clusters, a new node is provisioned automatically.

    At this point, Rancher will indicate that restoration from etcd snapshot is required.

1. Restore from an etcd snapshot.

    :::note

>     As the etcd node is a clean node, the path (`/var/lib/rancher/<k3s/rke2>/server/db/snapshots/`) may need to be manually created.

Reviewer: ```suggestion
    As the etcd node is a clean node, you may need to manually create the `/var/lib/rancher/<k3s/rke2>/server/db/snapshots/` path.
```

Author: Updated with suggestion

    :::

    - For S3 snapshots, restore using the UI.
      1. Click the **Snapshots** tab to view the list of saved snapshots.
      1. Go to the snapshot you want to restore and click **⋮ > Restore**.
      1. Select a **Restore Type**.
      1. Click **Restore**.
    - For local snapshots, restore using the UI is **not** available.
      1. In the upper right corner, click **⋮ > Edit YAML** to configure the setting `spec.cluster.rkeConfig.etcdSnapshotRestore.name` as the filename of the snapshot on disk in `/var/lib/rancher/<k3s/rke2>/server/db/snapshots/`.
      1. Example YAML that can be added under your `rkeConfig` configuration:

      ```yaml
        ...
        rkeConfig:

>             etcdSnapshotRestore:
>                 name: <string> #Refers to the filename of the associated etcdsnapshot object.
>                 generation: <int> #Changing the generation initiates a snapshot restore.
>                 restoreRKEConfig: <string> #Set to either none (or empty string), all, or kubernetesVersion.

Reviewer 2: ```suggestion
          etcdSnapshotRestore:
              name: <string> # Refers to the filename of the associated etcdsnapshot object.
              generation: <int> # Changing the generation initiates a snapshot restore.
              restoreRKEConfig: <string> # Set to either none (or empty string), all, or kubernetesVersion.
```

The YAML via 'Edit YAML' uses 2 space indentation.

Are `generation` and `restoreRKEConfig` required here?

Author: Just updated after getting some clarification about the [requirements](https://github.com/rancher/rancher-docs/issues/1246#issuecomment-2102129449), only the `name` field is necessary. I ended up removing the other optional fields to reduce any confusion.

        ...

>       ```

Reviewer: Maybe keep the button click and the description of the thing to configure as two distinct steps, but combine line 103 as part of the second step?

Author: Updated this section with suggestion and split the steps, also removed the "configure the setting `spec.cluster.rkeConfig.etcdSnapshotRestore.name` as the filename of the snapshot on disk in `/var/lib/rancher/<k3s/rke2>/server/db/snapshots/`" as the example YAML replaces this information.
